packages/dev/docs/pages/blog/building-a-combobox.mdx

```diff
@@ -0,0 +1,128 @@
+
+<Image src={screenreaderSpreadsheetImageUrl} alt="Screenshot of ComboBox screen reader vs browser test matrix" expandable />
+
+#### NVDA
+
+Initially, our ComboBox would automatically focus the first item in the menu whenever opened so the user wouldn't have to issue another keypress to begin menu navigation. However, when we tested the ComboBox using NVDA we discovered that
+character deletions and text cursor movement in the ComboBox input weren't being announced at all. NVDA would only resume announcing if no menu items had virtual focus via `aria-activedescendant`, so we ended up clearing menu item focus on any changes to the input text or
+left/right arrow key presses.
+
+#### VoiceOver
+
+VoiceOver has limited support for `aria-activedescendant`, failing to announce when the focused ComboBox item changed in a variety of situations. In Safari, changes to the `aria-activedescendant` aren't announced
+when the input is empty. In Chrome, VoiceOver only announces changes to `aria-activedescendant` if `aria-selected="true"` is applied to the focused item. If the list of options is organized into sections using `role="group"`,
+VoiceOver doesn't announce the focused item at all in any browser.
+
+Announcing item focus wasn't the only issue we encountered. VoiceOver doesn't announce the number of options currently available in the menu or the current section title, information that an end user would find helpful when navigating the ComboBox menu.
+When selecting an option from the menu, an announcement of the item's name is made but the fact that it is now "selected" is omitted.
+
+As a workaround, we made use of [ARIA live-regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions). When the ComboBox's menu is opened for the first time, we create two visually hidden `aria-live` regions in the DOM with two `div` elements, one with `aria-live="polite"` and the other `aria-live="assertive"`
+so that we have control over how the message should be announced to the user. Then whenever updates occur in the ComboBox, we can have the screen reader announce custom messages by updating the contents of said `div` elements. Thankfully, the bulk of this functionality was setup already in
+React Aria's `LiveAnnouncer`, allowing us to reuse it for our ComboBox after some updates.
+
+Special care was taken such that the messages themselves only contained relevant and concise information tailored to the current state of the ComboBox at any given time. For example, the ComboBox only announces the current section name and item count when the user enters a new section in the menu.
+When the user then moves to a different option in the same section, only the newly focused item name is announced. Similarly, the total option count is only announced when number of options available in the menu changes. Since many of these messages were added to fill in gaps in VoiceOver's announcement,
```

Comment:
```suggestion
When the user then moves to a different option in the same section, only the newly focused item name is announced. Similarly, the total option count is only announced when number of options available in the listbox changes. Since many of these messages were added to fill in gaps in VoiceOver's announcement,
```